src/v/config/base_property.h

```diff
@@ -77,6 +83,23 @@ class base_property {
     virtual void reset() = 0;
     virtual bool is_default() const = 0;
 
+    /**
+     * Helper for logging string-ized values of a property, e.g.
+     * while processing an API request or loading from file, before
+     * the property itself is initialized.
+     *
+     * Use this to ensure that any logged values are properly
+     * redacted if secret.
+     */
+    template<typename U>
+    std::string_view log_value(U const& in) {
```

Comment:
This is a slightly strange api, since the property exists, with a value, but I assume there's some strange YAML-based reason?

Reply:
Right, this is for when we are dealing with the raw inputs before they've been assigned into the property -- we need to be able to log at that point, because in many cases (esp errors) the value never hits the actual property's _value.  It's sort of spiritually a static method, but can't be because relies on the instance's metadata object to check if it's secret.

I've updated the name to format_raw to make it sound less like it logs the value (as it doesn't log and what it logs isn't the current value!)